src/main/java/com/thelow_quest_helper/thelow_quest_helper/item/Keyclick.java

```diff
@@ -70,13 +74,85 @@ public void onTick(TickEvent.ClientTickEvent event) {
     				String info = Town.getNearestTownInfo(d.x, d.y, d.z);
     				mc.thePlayer.addChatMessage(new ChatComponentText("§a[thelow_quest_helper]§7マーカーを設置しました§e("+d.x+","+d.y+","+d.z+")"));
     				Minecraft.getMinecraft().thePlayer.addChatMessage(new ChatComponentText(info));
-    				clicked = true;
+    				wkeyClicked = true;
     				ItemHoverTracker.lastLore = null;
                 }
             }
             
+        }else if(Keyboard.isKeyDown(Keyboard.KEY_Z)&&!zkeyClicked) {
+            List<String> lastLore = ItemHoverTracker.lastLore;
+            if (lastLore == null) return;
+
+            for (String line : lastLore) {
+                String clean = line.replaceAll("§.", "").trim();
+                
+                int x1 = getPlayerBlockPos().getX(), y1 = getPlayerBlockPos().getY(), z1 = getPlayerBlockPos().getZ();
+                String[] routeInfo = null;
+                BestTeleport_mtb routeCalculator = new BestTeleport_mtb();
+                
+                
+                if(clean.contains("地上世界")) {
+                	Matcher matcher = Pattern.compile("\\((-?[0-9.]+), (-?[0-9.]+), (-?[0-9.]+)\\)").matcher(clean);
+                    if (matcher.find()) {
+                        int x = Integer.parseInt(matcher.group(1));
```

Comment:
座標をintで取得していますが、NPCの座標はdoubleで記述しているので、doubleで取得した後にintにキャストした方がいです
```suggestion
                        int x = (int) Double.parseDouble(matcher.group(1));
```